cmd/clusterctl/client/cluster/components.go

```diff
@@ -170,16 +167,17 @@ func (p *providerComponents) Delete(options DeleteOptions) error {
 			namespacesToDelete.Insert(obj.GetName())
 		}
 
-		// If not a shared resource or not a namespace
-		if !isSharedResource && !isNamespace {
-			// If the resource is a cluster resource, skip it if the resource name does not start with the instance prefix.
-			// This is required because there are cluster resources like e.g. ClusterRoles and ClusterRoleBinding, which are instance specific;
-			// During the installation, clusterctl adds the instance namespace prefix to such resources (see fixRBAC), and so we can rely
-			// on that for deleting only the global resources belonging the the instance we are processing.
-			if util.IsClusterResource(obj.GetKind()) {
-				if !strings.HasPrefix(obj.GetName(), instanceNamespacePrefix) {
-					continue
-				}
+		// If the resource is a cluster resource, skip it if the resource name does not start with the instance prefix.
+		// This is required because there are cluster resources like e.g. ClusterRoles and ClusterRoleBinding, which are instance specific;
+		// During the installation, clusterctl adds the instance namespace prefix to such resources (see fixRBAC), and so we can rely
+		// on that for deleting only the global resources belonging the the instance we are processing.
+		// NOTE: namespace and CRD are special case managed above; webhook instead goes hand by hand with the controller they
+		// should always be deleted.
+		isWebhook := obj.GroupVersionKind().Kind == validatingWebhookConfigurationKind || obj.GroupVersionKind().Kind == mutatingWebhookConfigurationKind
+		if util.IsClusterResource(obj.GetKind()) &&
+			!(isNamespace || isCrd || isWebhook) {
+			if !strings.HasPrefix(obj.GetName(), instanceNamespacePrefix) {
+				continue
 			}
 		}
```

Comment:
```suggestion
		if util.IsClusterResource(obj.GetKind()) &&
			!isNamespace && !isCrd && !isWebhook &&
			!strings.HasPrefix(obj.GetName(), instanceNamespacePrefix) {
			continue
		}
```
Maybe that's easier to comprehend (hope it's equivalent)

Reply:
done!